### PR TITLE
Distinguish flex flow and dag flow(use flow.flex.yaml for flex flow, flow.dag.yaml for dag flow)

### PR DESCRIPTION
**Pull request by Stephen1993**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

It is necessary to distinguish between the flow yaml used by the eager flow and the dag flow (flex yaml and dag yaml), and modify the code to enable the eager flow to use flex yaml correctly and to enable the code to automatically recognize flex yaml and dag flex, which is divided into two steps:

1. Distinguishing between flow.flex.yaml and flow.dag.yaml is what this PR is doing.
2. Modify the code and migrate eager flow to flex flow. This [PR ](https://github.com/microsoft/promptflow/pull/2548/files#diff-a56ebf7117cd090f1115975b561939d4b6b66df2a3c0b92dece68e17054b1e9e)is currently working on this.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/_utils/flow_utils.py [hunk 1]
```diff
@@ -70,9 +73,8 @@ def resolve_flow_path(
     :param base_path: The base path to resolve the flow path. If not specified, the flow path will be
       resolved based on the current working directory.
     :type base_path: Union[str, Path, PathLike]
-    :param new: If True, the function will return the flow directory path and the file name of the
-        target yaml that should be created. If False, the function will try to find the existing
-        target yaml and raise FileNotFoundError if not found.
+    :param check_flow_exist: If True, the function will return the flow directory path and the file name of the
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/_utils/flow_utils.py:76]: If True, the function will try to check the target yaml and raise FileNotFoundError if not found?

File: src/promptflow-core/promptflow/_utils/flow_utils.py [hunk 2]
```diff
@@ -81,17 +83,38 @@ def resolve_flow_path(
...
-            return flow_path, DEFAULT_FLOW_YAML_FILE_NAME
-        return flow_path.parent, flow_path.name
+    if flow_path.is_dir():
+        target_folder = flow_path
+        dag_file_exist = (target_folder / FLOW_DAG_YAML).exists()
+        flex_file_exist = (target_folder / FLOW_FLEX_YAML).exists()
+        target_file = FLOW_FLEX_YAML if flex_file_exist else FLOW_DAG_YAML
+        if dag_file_exist and flex_file_exist:
+            raise ValidationException(
+                f"Both exist {FLOW_DAG_YAML} and {FLOW_FLEX_YAML} in the flow path {flow_path}, "
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/_utils/flow_utils.py:93]: Both {FLOW_DAG_YAML} and {FLOW_FLEX_YAML} exist in {flow_path}. Please specify a file or remove the extra YAML.

Comment (Stephen1993) [reply src/promptflow-core/promptflow/_utils/flow_utils.py]: done

File: src/promptflow-core/promptflow/_utils/flow_utils.py (hunk 1 above)
Comment (Stephen1993) [reply src/promptflow-core/promptflow/_utils/flow_utils.py]: done

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/tests/sdk_cli_test/unittests/test_flow.py [hunk 3]
```diff
@@ -65,3 +66,13 @@ def test_flow_load_invalid(self, kwargs, error_message, exception_type):
             load_flow(**kwargs)
 
         assert error_message in str(e.value)
+
+    def test_mutiple_flow_load(self):
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_test/unittests/test_flow.py:70]: typo: `mutiple` -> `multiple`

Comment (Stephen1993) [reply src/promptflow/tests/sdk_cli_test/unittests/test_flow.py]: good catch, fix it.

**Review by 0mza987 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**